twittermap/public/javascripts/cache/services.js

```diff
@@ -0,0 +1,321 @@
+                                    [bounds._southWest.lng,bounds._southWest.lat],
+                                    [bounds._southWest.lng,bounds._northEast.lat],
+                                    [bounds._northEast.lng,bounds._northEast.lat]
+                                ]]);
+
+       var bbox = turf.bbox(currentRequestPolygon);
+        // to search in Rbush Tree ,we need the MBR of the requested region.
+       var item = {
+           minX : bbox[0],
+           minY : bbox[1],
+           maxX : bbox[2],
+           maxY : bbox[3]
+       }
+
+
+       if(typeof cachedRegion != "undefined" && typeof turf.difference(currentRequestPolygon,cachedRegion) == "undefined") {
+              //cache HIT
+
+             var result = cachedCityPolygonTree.search(item);
+             data_response = turf.featureCollection(result);
+             deferred.resolve(data_response);
+
+             return deferred.promise();
+
+       }else{
+                   //cache MISS
+
+              var buffered = turf.buffer(currentRequestPolygon, preFetchDistance , 'miles');
+              var bboxBuffer = turf.bbox(buffered);
+                   //Pre Fetch
+              var rteExtends = "city/" + bboxBuffer[3] + "/" + bboxBuffer[1] + "/" + bboxBuffer[2] + "/" + bboxBuffer[0];
+
+              $http.get(rteExtends).success(function(data) {
+
+                               var result_set = insertIntoTree(data.features,currentRequestPolygon).done(function(){
+
+                                                var extendPoly = turf.bboxPolygon(bboxBuffer);
+
+                                                if(cachedRegion == undefined)
+                                                    {cachedRegion = extendPoly;}
+                                                else
+                                                    {cachedRegion = turf.union(extendPoly,cachedRegion);}
+
+                                                var result = cachedCityPolygonTree.search(item);
+                                                data_response = turf.featureCollection(result);
+
+                                                deferred.resolve(data_response);
+
+
+                      });
+              }).error(function(data) {
+                                    console.error("Load city data failure");
+              });
+              return deferred.promise();
+       }
+  }
+//to insert polygons into the Rtree,features are the city polygons
+ var insertIntoTree = function insertIntoTree(features,currentRequest){
+
+       var deferred = new $.Deferred();
+       var nodes = [];
+       var treeID;
+
+       for(var id in features){
+
+           var box = turf.bbox(features[id]);
+           features[id].minX = box[0];
+           features[id].minY = box[1];
+           features[id].maxX = box[2];
+           features[id].maxY = box[3];
+           features[id].properties["centerLog"] = (features[id].maxX + features[id].minX) / 2;
```

Comment:
It is a field we have in format of our geo JSONs .Could you see the loadCityJsonByBound I have removed setCenterAndBoundary and written a comment there,This function does the same work as setCenterAndBoundary does along with what is needed for Cache.We talked about this some time ago.